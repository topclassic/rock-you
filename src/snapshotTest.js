
import renderer from 'react-test-renderer'

const snapshotTest = (testCase, element) => {
  it(testCase, () => {
    const wrapper = renderer.create(element).toJSON()
    expect(wrapper).toMatchSnapshot()
  })
}

export default snapshotTest