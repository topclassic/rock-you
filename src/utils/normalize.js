export const normalizeDigit = (value) => {
    if (!value) {
      return value
    }
    return value.replace(/[^\d.]/g, '')
  }