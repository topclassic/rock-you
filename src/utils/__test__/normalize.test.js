import { normalizeDigit } from "../normalize";
import snapshotTest from "../../snapshotTest";

describe("NormalizeDigit", () => {
  it("test func normalizeDigit correctly", () => {
    expect(normalizeDigit("100")).toEqual("100");
  });
  it("test func normalizeDigit not correctly", () => {
    expect(normalizeDigit("")).toEqual("");
    expect(normalizeDigit("A")).toEqual("");
  });

});

describe("NormalizeDigit testing", () => {
  snapshotTest("func normalizeDigit(100)", normalizeDigit("100"));
  snapshotTest("func normalizeDigit('')", normalizeDigit(""));
  snapshotTest("func normalizeDigit(A)", normalizeDigit(""));
});
