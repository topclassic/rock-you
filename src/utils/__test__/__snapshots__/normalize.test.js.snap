// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NormalizeDigit testing func normalizeDigit('') 1`] = `""`;

exports[`NormalizeDigit testing func normalizeDigit(100) 1`] = `"100"`;

exports[`NormalizeDigit testing func normalizeDigit(A) 1`] = `""`;
